packages/flutter/test/gestures/mouse_tracking_test.dart

```diff
@@ -350,16 +380,161 @@ void main() {
     ]));
     events.clear();
 
-    // Detach the annotation
+    // Detaching an annotation should not trigger events
     isInHitRegion = false;
     _mouseTracker.detachAnnotation(annotation);
+    expect(events, _equalToEventsOnCriticalFields(<PointerEvent>[
+    ]));
+    expect(_binding.postFrameCallbacks, hasLength(0));
+
+    _binding.scheduleMouseTrackerPostFrameCheck();
+    expect(_binding.postFrameCallbacks, hasLength(1));
+
+    _binding.flushPostFrameCallbacks(Duration.zero);
+    expect(events, _equalToEventsOnCriticalFields(<PointerEvent>[
+    ]));
+    expect(_binding.postFrameCallbacks, hasLength(0));
+  });
+
+  test('should correctly handle when annotation moves in or out of the pointer', () {
+    bool isInHitRegion;
+    final List<Object> events = <PointerEvent>[];
+    final MouseTrackerAnnotation annotation = MouseTrackerAnnotation(
+      onEnter: (PointerEnterEvent event) => events.add(event),
+      onHover: (PointerHoverEvent event) => events.add(event),
+      onExit: (PointerExitEvent event) => events.add(event),
+    );
+    _setUpMouseAnnotationFinder((Offset position) sync* {
+      if (isInHitRegion) {
+        yield annotation;
+      }
+    });
+
+    // Start with an annotation attached
```

Comment:
```suggestion
    // Start with an annotation attached.
```

In general, all comments should be complete sentences with punctuation:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-correct-grammar